src/Microsoft.ML.Data/Transforms/RowShufflingTransformer.cs

```diff
@@ -594,10 +594,10 @@ private async Task ProduceAsync()
                 try
                 {
                     int circularIndex = 0;
-                    while (await _toProduce.OutputAvailableAsync().ConfigureAwait(false))
+                    while (await _toProduceChannel.Reader.WaitToReadAsync().ConfigureAwait(false))
```

Comment:
Not too sure if this is the best replacement for `OutputAvailableAsync` within the Channel.